code/game/objects/items/stacks/stack.dm

```diff
@@ -355,7 +357,7 @@
 	use(amount)
 
 /obj/item/stack/attackby(obj/item/W, mob/user, params)
-	if(istype(W, merge_type))
+	if(!parent_stack && istype(W, merge_type) || parent_stack && W.type == type)
```

Comment:
please use more brackets here, I'm not 100% sure what this if check is doing

Reply:
Sorry for getting back to this so late, I was out of commission last week. The way (I think) this works is if the parent_stack flag is false, then it merges like a normal stack. If the flag is true, then it merges only with A) other items with the flag, and B) have an identical type as the one attempting to be stacked.